cdi/src/main/java/org/hibernate/validator/cdi/spi/InjectingConstraintValidatorFactory.java

```diff
@@ -39,6 +45,14 @@ public InjectingConstraintValidatorFactory(BeanManager beanManager) {
 
 	@Override
 	public <T extends ConstraintValidator<?, ?>> T getInstance(Class<T> key) {
+		// we can do a simple check like :
+		//		if ( key.getPackageName().startsWith( "org.hibernate.validator.internal.constraintvalidators.bv" )
+		//				|| key.getPackageName().startsWith( "org.hibernate.validator.internal.constraintvalidators.hv" ) ) {
+		//			return super.getInstance( key );
+		//		}
```

Comment:
Yeah so all things considered, this looks like a better solution. I'd use a simple `startsWith("org.hibernate.validator.internal.constraintvalidators.")` though.
And you'll need some test to check that all built-in constraint validators indeed are in that package...